### PR TITLE
Add --version and create version at compile time

### PR DESCRIPTION
**Pull request by KlausVii**
Issue #35 
version is created from 

```
git describe --tags
```

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+VERSION := $(shell git describe --tags)
+# There are the default architecture values can be changed via 'make OS=linux ARCH=amd64'
+OS = darwin
```
Comment (jeremyletang) [inline makefile]: Maybe we can remove them, and not specify the OS / ARCH for the all build so you can just use make and build the binary for your current platform.
Moreover we could add a release rule in the makefile which could build for each targets we support with borg.
Do not forget to add the -w linker flag in order to remove the DWARF debug symbols from the releases binaries.
And could you add a call to upx for each binary you generate in order to make them smaller.

**Comment by jeremyletang** [comment]
oh moreover it could be intersting to include the git sha1 of the commit of the build in the version. So if the user with a bug built borg himself, we can know with which exact version it was.
`borg version 1.2.0 (linux/amd64 cab3bb14facc6eb6985e13f8a49ce3c662fa7d06)`.

**Comment by KlausVii** [comment]
`git describe --tags` outputs: `v0.0.2-50-g533652d`, where 50 is the number of commits since the v0.0.2 and `533652d` is the short SHA of the latest commit, so it already contains the exact version info.

File: makefile (hunk 1 above)
Comment (KlausVii) [reply makefile]: So the normal make would not have OS/ARCH? I could also just print them from `uname` for the all build, though this is not exactly the same format used by GOOS/GOARCH.
What's this -w linker business?
I think I get rest.

**Comment by jeremyletang** [comment]
hey this is working well, thanks for the PR !

**Pull request merged**